### DISCUSSION TITLE
Can I edit the models_gen.go file to add gorm clause and tags?

### DISCUSSION THREAD

**Discussion by Mrbeyond** [99designs/gqlgen#2563] [Q&A]
Is it allowed to manually edit models_gen file to add gorm tags to the generated model?

Example:

Exact model I need:

type User struct {
	ID          uint64         `json:"-" gorm:"primaryKey;autoIncrement"`
	Email       string         `json:"email" binding:"email,required" gorm:"unique; not null"`
	FirstName   string         `json:"first_name"
	LastName    string         `json:"last_name"
	Country     string         `json:"country"
	CreatedAt   time.Time      `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt   time.Time      `json:"updated_at" gorm:"autoUpdateTime"`
	DeletedAt   gorm.DeletedAt `json:"deleted_at"`
}

Model generated:

type User struct {
	ID          string             `json:"id"`
	Email       string             `json:"email"`
	FirstName   *string            `json:"first_name"`
	LastName    *string            `json:"last_name"`
	Country     *string            `json:"country"`
	CreatedAt   *time.Time         `json:"created_at"`
	UpdatedAt   *time.Time         `json:"updated_at"`
	DeletedAt   *time.Time         `json:"deleted_at"`
}

Please I need answer to this, thanks.

**Comment by it512** [comment]
if using vim and vim-go plugin , you can try command :GoAddTags gorm ,

**Reply by Mrbeyond** [reply]
I use VScode.

Any approach around this?

**Reply by it512** [reply]
yes， vscode can use it yet 
see。  https://github.com/fatih/gomodifytags
and     https://github.com/golang/vscode-go

**Reply by Mrbeyond** [reply]
@it512 thanks for your help, I really appreciate this. I have checked it out.

So how can I use gomidifytags with gqlgen schema?

Because even if I add the tags manually, re-generating removes them all and I would have to add ti all over again.

**Comment by it512** [comment]
edit your gqlgen.yml
to add 
struct_tag: gorm

**Reply by Mrbeyond** [reply]
Can you please help explain how to achieve this?

**Reply by pk0205** [reply]
I generally split the models related to the table in different files like user.go and then in gqlgen.yml add

```
models:
  User:
    model: github.com/.../graph/model.User
```

This way generate command won't re-generate the User model and you can add all the gorm tags there
